x-pack/osquerybeat/internal/osqdcli/client.go

```diff
@@ -50,8 +54,13 @@ func (e *ErrorQueryFailure) Error() string {
 }
 
 type Client struct {
-	socketPath     string
+	socketPath string
+
+	// Query timeout, currently can only be set at the transport level.
+	// This means that while the query will return with error the osqueryd internally continues to execute the query until completion.
+	// This is a known issue with osquery/osquery-go/thrift RPC implementation at the moment: there is effectively no way to cancel the long running query
```

Comment:
Is there an issue link for this? If there is, it would be helpful to include it in comment.

Reply:
There is no open issue for that in osquery repository (that's the thirdparty software that is used here). I was communicating with osquery/osquery-go developers about that, they are aware of this issue. 
Sent them repro example https://github.com/aleksmaus/osqlong showing the issue.
Maybe when we have time we could dig to figure out if this is fixable and contribute back, not sure about that yet.